alas_but_one.py

```diff
@@ -1,39 +1,85 @@
 import os
 import re
-from collections import Counter
+from spellchecker import SpellChecker
+
+from collections import defaultdict
+import string
 
 # Usage
 # -----
 # This script reads files in the relative "source/" directory and sub-directories
 # named with the "rst" or "txt" extension and counts the instances of each
-# word (as matched by the WORD_REGEX constant).
-# Update the MAX_OCCUR=n constant to print out only words that are repeated
+# word.
+# Update the occur_freq=n constant to print out only words that are repeated
 # up to n times.
 #
 # Backstory
 # ---------
-# A user reported an issue with documentation in which "Atlas" was misspelled 
-# as "Alas". After investigation, a writer reported there was only one instance 
+# A user reported an issue with documentation in which "Atlas" was misspelled
+# as "Alas". After investigation, a writer reported there was only one instance
 # of the type. Hence, "there was, alas, but one".
 
-MAX_OCCUR=1
-WORD_REGEX=re.compile("[\w'_\-]+")
 
-directory = 'source'
-filename_re = r'\.(txt|rst)$'
+def count_in_dir():
+    directory = "source"
+    filename_re = r"\.(txt|rst)$"
+
+    processed = []
+    words_and_freqs = []
+
+    for root, dirs, files in os.walk(directory):
+        for filename in files:
+            if re.search(filename_re, filename):
+                filepath = os.path.join(root, filename)
+                # print(filepath)
+                read_file = open(filepath, "r", encoding="utf-8")
+                # processed = []
+
+                for line in read_file:
+                    line = line.strip()
+                    words = re.split(r"\s*(?:[()]|'s|\s|-|`|/|#)\s*", line)
```

Comment:
Since this iterates per line, the regex potentially needs to get built every time. Instead, pull this outside of the loop and use re.compile() to instruct the parser to keep it ready for reuse.